### PR TITLE
Fix endianness in result.mitigator

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

```python
from qiskit.result import LocalReadoutMitigator, CorrelatedReadoutMitigator
from qiskit import Aer, QuantumCircuit
from qiskit.test.mock import FakeBogota

backend = Aer.get_backend("aer_simulator").from_backend(FakeBogota())
mitigator = LocalReadoutMitigator(backend=backend, qubits=[0, 1])
circ = QuantumCircuit(2)
circ.x(0)
circ.measure_all()
counts = backend.run(circ).result().get_counts()
print(mitigator.expectation_value(counts, diagonal="IZ"))
```

This expectation value is 1 in current main branch. However, it should be -1 since Qiskit is little endian.
For example, quantum_info module outputs it is -1 such as
```python
from qiskit.quantum_info import Statevector, Pauli
circ = QuantumCircuit(2)
circ.x(0)

Statevector(circ).expectation_value(Pauli("IZ"))
```
.

### Details and comments

I'll add the release note after this fix is confirmed to be valid.

### CODE REVIEW

**Comment by gadial** [comment]
Thanks @ikkoham, this looks good.

**Review by ShellyGarion — APPROVED** [review]
LGTM. Thanks @ikkoham !

**Review by jakelishman — COMMENTED** [review]
This needs a bugfix release note, but otherwise looks good.

File: test/python/result/test_mitigators.py [hunk 1]
```diff
@@ -431,6 +434,15 @@ def test_error_handling(self):
             "Assignment matrix columns must be valid probability distributions",
         )
 
+    def test_expectation_value_endian(self):
+        """Test that endian for expval is little."""
+        mitigators = self.mitigators(self.assignment_matrices())
+        counts = Counts({"10": 3, "11": 24, "00": 74, "01": 923})
+        for mitigator in mitigators:
+            expval, _ = mitigator.expectation_value(counts, diagonal="IZ", qubits=[0, 1])
+            print(expval)
```
Comment (jakelishman) [inline test/python/result/test_mitigators.py:443]:
```suggestion
```

**Review by jakelishman — APPROVED** [review]

**Pull request merged**